### DISCUSSION TITLE
Modifying the checkbox component

### DISCUSSION THREAD

**Discussion by Fazle-Jahangir** [twilio-labs/paste#2414] [General] [answered]
**Background**

As previously discussed in  #2332 I am adding a new feature to _Insights Queue Stats_ where users are able to hide individual columns in a table by deselecting checkboxes in a side modal/panel and vice versa for showing them. 

**Questions**

The typographical hierarchy of the **checkbox group** component (second attached image) has been called into question by the design team at Flex due to the overuse of font weights on the heavier side( e.g., bold). I have taken the liberty in changing the font weight in my design (first attached image) so that only the **checkbox group label** is in bold, the **select all** checkbox is medium weight and the remaining checkbox labels are regular weight. How do you feel or do you have any opinions regarding these changes? 

Secondly, the **select all** checkbox is not aligned with the other checkboxes, aligned slightly more to the left. I don't have a strong preference for either one but I am interested to hear your thoughts on aligning **select all** with the rest. 

<img width="423" alt="Screenshot 2022-05-18 at 15 49 05" src="https://user-images.githubusercontent.com/100706574/169070960-2070997e-4a21-4482-86c1-39ee2f66782d.png">
<img width="297" alt="Screenshot 2022-05-18 at 15 48 38" src="https://user-images.githubusercontent.com/100706574/169070995-d23c2a09-b27f-4026-a669-6016cea0c4e3.png">

**Comment by nkrantz** [comment, marked as answer]
Hey @Fazle-Jahangir - the current design of Checkbox (and Checkbox Group) has already gone through peer review which means it was approved by the Flex design team. Though the design is finalized, there's still room to make updates if you can show some evidence that the updates would improve the customer experience in some way. Could you provide a little more info on why the current design is being questioned? Have you received customer feedback or done any testing that showed usability issues with the current Checkbox Group? Another thing that would be helpful is showing a comparison of 2 layouts with the 2 different font-weights and alignments to show the difference within a UI.

In order to make the changes you've proposed here, we'll need to go through the review process with your updated design to ensure that there will be a significant improvement from the current design. That would include presenting the new design to the Design Systems Committee along with your reasoning for making the changes. Feel free to set up a meeting with me to go over how to best prep for that and to choose a date for the Committee meeting!

**Reply by nkrantz** [reply]
Checking in here @Fazle-Jahangir - how did you decide to move forward with the Checkbox design?

**Reply by Fazle-Jahangir** [reply]
Hi @nkrantz, I have taken your feedback into account and will add the checkbox group to my user testing planned in the near future. If there is any substantial customer feedback I will share the findings with you and we can decide if it's worth having another look at the design. At the moment it is just subjective viewpoints from the Flex team on the usage of bold and medium font weights not providing a clear typographical hierarchy. Also, the "select all" checkbox being aligned slightly to the left of the rest of the group. However, as you've said it's worth to first explore customer feedback on this.

**Reply by nkrantz** [reply]
Thanks for the update, @Fazle-Jahangir! I'm going to remove the `Active Discussion` label for now but feel free to update this thread whenever you'd like to continue the conversation.
